expression/builtin_control.go

```diff
@@ -53,6 +53,67 @@ type caseWhenFunctionClass struct {
 	baseFunctionClass
 }
 
+// Infer result type for builtin IF, IFNULL && NULLIF.
+func inferType(tp1, tp2 *types.FieldType) *types.FieldType {
```

Comment:
to avoid name conflicts in package `expression`: s/inferType/inferType4ControlFuncs/